yarn-project/circuit-types/src/l2_block_downloader/l2_block_stream.ts

```diff
@@ -58,12 +58,12 @@ export class L2BlockStream {
       });
 
       // Check if there was a reorg and emit a chain-pruned event if so.
-      let latestBlockNumber = localTips.latest;
+      let latestBlockNumber = localTips.latest.number;
       while (!(await this.areBlockHashesEqual(latestBlockNumber, sourceTips.latest))) {
```

Comment:
Not an issue introduced in this PR. But does this block stream assume that if a re-org happens then the local latest block number will be > source latest? Could we have a situation where a re-org occurs and then there is a new pending chain that is ahead of the local latest block number? 

This loop looks like it would just keep going back to block 0, but maybe I am not reading it right,.

Reply:
Yeah this definitely looks possible to me. 

If we get part way through epoch 0, then shut off the node, then the chain reorgs to genesis, but builds ahead a ways, then when we start the node back up, I think this will re-org to zero.

I filed https://github.com/AztecProtocol/aztec-packages/issues/9314

Reply:
That should not be the case. `areBlockHashesEqual` tests whether local and source agree on the block hash at a given height (the first param) and accepts a "cache" of data already requested from source to avoid re-requesting it. 

I can see how the function naming and signature (with both args at the same "level") make it easy to misunderstand though. I've pushed this PR to make it clearer: https://github.com/AztecProtocol/aztec-packages/pull/9481/files.